### ISSUE TITLE
if i want to use the d_loss on caffe , how can i compute the Gradient

### ISSUE THREAD

**Issue by paleblackless** [carpedm20/BEGAN-tensorflow#50]
i want to use the Wasserstein loss on caffe,as we know BEGAN use  tf.train.AdamOptimizer() to min the loss , however i want to to use the Wasserstein loss that 'd_loss =|Y-D(Y)| - |D(G(z)-G(z))|' on the caffe. how to compute the Gradient
